packages/docusaurus-mdx-loader/src/remark/toc/index.ts

```diff
@@ -7,41 +7,42 @@
 
 import {parse, type ParserOptions} from '@babel/parser';
 import traverse from '@babel/traverse';
-import stringifyObject from 'stringify-object';
 import {toValue} from '../utils';
+import {hasImports, isExport, isImport} from './utils';
+import type {SpreadElement} from 'estree';
 import type {Identifier} from '@babel/types';
 import type {Node, Parent} from 'unist';
 import type {Heading, Literal} from 'mdast';
 // @ts-expect-error: TODO see https://github.com/microsoft/TypeScript/issues/49721
 import type {Transformer} from 'unified';
 import type {
   MdxjsEsm,
+  MdxJsxFlowElement,
   // @ts-expect-error: TODO see https://github.com/microsoft/TypeScript/issues/49721
 } from 'mdast-util-mdx';
 
-// TODO as of April 2023, no way to import/re-export this ESM type easily :/
-// TODO upgrade to TS 5.3
-// See https://github.com/microsoft/TypeScript/issues/49721#issuecomment-1517839391
-// import type {Plugin} from 'unified';
-type Plugin = any; // TODO fix this asap
-
 export type TOCItem = {
   readonly value: string;
   readonly id: string;
   readonly level: number;
 };
 
+type NestedTOC = {
+  readonly nested: true;
+  readonly name: string;
+};
+
+// TODO as of April 2023, no way to import/re-export this ESM type easily :/
+// TODO upgrade to TS 5.3
+// See https://github.com/microsoft/TypeScript/issues/49721#issuecomment-1517839391
+// import type {Plugin} from 'unified';
+type Plugin = any;
```

Comment:
Was confused to see a diff on this. Move it at the top above TOCItem like it was before to remove the useless diff?